hall_manager/models.py

```diff
@@ -1,5 +1,61 @@
 from django.db import models
 
+class Slot(models.Model):
+    TIMING_CHOICES = [
```

Comment:
I was thinking of introducing serializers to validate these checks before it reaches the db. This can be an action item for all the other implementations as well